src/generators/AbstractGenerator.ts

```diff
@@ -34,28 +34,58 @@ export abstract class AbstractGenerator<Options extends CommonGeneratorOptions =
   }
 
   public abstract render(model: CommonModel, inputModel: CommonInputModel): Promise<RenderOutput>;
+  public abstract renderFull(model: CommonModel, inputModel: CommonInputModel, options: RenderFullOptions): Promise<RenderOutput>;
 
   public process(input: Record<string, unknown>): Promise<CommonInputModel> {
     return InputProcessor.processor.process(input, this.options.processorOptions);
   }
 
-  public async generate(input: Record<string, unknown> | CommonInputModel): Promise<OutputModel[]> {
-    if (input instanceof CommonInputModel) {
-      return this.generateModels(input);
-    }
-    const model = await this.process(input);
-    return this.generateModels(model);
+  /**
+   * Generates the full output of a model, instead of a scattered model.
+   * 
+   * OutputModels result is no longer the model itself, but including package, package dependencies and model dependencies.
+   * 
+   * @param input 
+   * @param options to use for rendering full output
+   */
+  public async generateFull(input: Record<string, unknown> | CommonInputModel, options: RenderFullOptions): Promise<OutputModel[]> {
```

Comment:
I still can't figure out if this is the `right` name for the function 🤷

Reply:
Maybe `generateFullContent`?

Reply:
Ok, it can be :)